general/socketstream.cpp

```diff
@@ -146,7 +146,8 @@ int socketbuf::open(const char hostname[], int port)
       }
 #endif
 
-      if (connect(socket_descriptor, rp->ai_addr, rp->ai_addrlen) < 0)
+      if (connect(socket_descriptor, rp->ai_addr,
+                  static_cast<int>(rp->ai_addrlen)) < 0)
```

Comment:
As in the case of `bind` above, the cast in last argument should be to `socklen_t`. Since `socklen_t` is not defined with MSVC, we'll need to add a definition similar to `isockstream.cpp`:
```c++
typedef int socklen_t;
```
e.g. after this line: https://github.com/mfem/mfem/blob/7c1a0eb5e3f70ff9d161c9d6e3f228ba2513acca/general/socketstream.cpp#L32

Reply:
Or, after this line? https://github.com/mfem/mfem/blob/7c1a0eb5e3f70ff9d161c9d6e3f228ba2513acca/general/socketstream.cpp#L35

Reply:
The `typedef` should be inside the `#ifdef _MSC_VER`, so it is good as it is now. Thanks!